### PR TITLE
spec: Add missing property in Snapshot

### PR DESCRIPTION
**Pull request by Aitozi**
Some fields in Snapshot are lost, add them back

### CODE REVIEW

File: crates/paimon/src/spec/snapshot.rs [hunk 1]
```diff
@@ -134,4 +144,38 @@ impl Snapshot {
...
+        self.commit_kind
+    }
+}
+
+/// Type of changes in this snapshot.
+///
+/// Impl Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/Snapshot.java#L506>.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum CommitKind {
```
Comment (QuakeWang) [inline crates/paimon/src/spec/snapshot.rs:172]: @Aitozi We do not need this enum, you can refer #11.

File: crates/paimon/src/spec/snapshot.rs [hunk 2]
```diff
@@ -39,6 +39,16 @@ pub struct Snapshot {
     #[builder(default = None)]
     index_manifest: Option<String>,
     commit_user: String,
+    // Mainly for snapshot deduplication.
+    //
+    // If multiple snapshots have the same commitIdentifier, reading from any of these snapshots
```
Comment (Aitozi) [inline crates/paimon/src/spec/snapshot.rs:44]: I have a question out of this PR, Currently, the overwrite (expire partition) share the same commit identifier  with the  current snapshots's commit identifier. So I think this comments may not right ? CC @JingsongLi

File: crates/paimon/src/spec/snapshot.rs (hunk 1 above)
Comment (Aitozi) [reply crates/paimon/src/spec/snapshot.rs:172]: Hi @QuakeWang I think that just means `log_offsets` ?

**Comment by QuakeWang** [comment]
It's my bad, when add the snapshot lost these field. Maybe I can add these in #39 with some tests. What do you think of this?

**Comment by Aitozi** [comment]
@QuakeWang Sure, thanks.

**Pull request closed without merging**